src/xdg.c

```diff
@@ -276,8 +276,10 @@ position_xdg_toplevel_view(struct view *view)
 		view->x = center_x - view->xdg_surface->current.geometry.width / 2;
 		view->y = center_y - view->xdg_surface->current.geometry.height / 2;
 	}
-	view->x += view->ssd.margin.left;
-	view->y += view->ssd.margin.top;
+
+	struct border margin = ssd_get_margin(view->ssd);
```

Comment:
I think this is a hot path in which case the additional function call might actually impact the perceived latency on low power systems. Not sure how to deal with this though, maybe we could cache it somewhere?